modules/azure-core-cpp/1.14.1/overlay/BUILD.bazel

```diff
@@ -0,0 +1,94 @@
+load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
+load("@rules_cc//cc:defs.bzl", "cc_library")
+load("transitions.bzl", "curl_with_openssl")
+
+package(default_visibility = ["//visibility:private"])
+
+bool_flag(
+    name = "build_transport_curl",
+    build_setting_default = True,
+    visibility = ["//visibility:public"],
+)
+
+config_setting(
+    name = "build_transport_curl_setting",
+    flag_values = {":build_transport_curl": "true"},
+)
+
+bool_flag(
+    name = "build_transport_winhttp",
+    build_setting_default = False,
+    visibility = ["//visibility:public"],
+)
+
+config_setting(
+    name = "build_transport_winhttp_setting",
+    flag_values = {":build_transport_winhttp": "true"},
+)
+
+# Build Curl with OpenSSL instead of BoringSSL (the default SSL lib for the Curl
+# on BCR).
+curl_with_openssl(
+    name = "curl_with_openssl",
```

Comment:
Is it safe for some `cc_binary` to link in both the regular and the transitioned curl? If not, it could be better to let the top-level binary choose the curl flavor and react to it in your library via a `select` on that `curl` setting. Transitions work best when they are either fully contained or applied at the top level.